tools/eslint-rules/rules/no-missing-jsx-pragma.spec.ts

```diff
@@ -0,0 +1,353 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+import { rule, RULE_NAME } from './no-missing-jsx-pragma';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: { ecmaFeatures: { jsx: true } },
+});
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    {
```

Comment:
Suggestion: add `name` prop to test cases or comment above the case options describing why the case considered valid/invalid

Reply:
I tweaked the existing comments for exotic anti-patterns that we might have in our codebase.

in general it should not be necessary to provide comments per test case.

regarding the `name` I try to avoid that as even [official TS-ESlint is not using it](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/tests/rules/no-empty-function.test.ts). The UX is not adding that much value as well IMO ⬇

<img width="641" alt="image" src="https://github.com/user-attachments/assets/a4aef884-279e-4da0-8997-dada3845e193">


thoughts?

Reply:
> regarding the name I try to avoid that as even [official TS-ESlint is not using it](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/tests/rules/no-empty-function.test.ts?rgh-link-date=2024-10-09T12%3A24%3A15Z).

Agree, that it doesn't improve UX much.

Regarding comments, they might not be necessary for simple scenarios, but when the rule depends on multiple conditions like in our case "runtime: classic/automatic", "pragma: no/jsxRuntime/jsxImportSource", "slotUsed: true/false" it could be a bit easier to understand what's going on there.

Here is an comments example from the [typescript-eslint repo](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/tests/rules/member-ordering/member-ordering-alphabetically-case-insensitive-order.test.ts#L11)

Reply:
added additional comments https://github.com/microsoft/fluentui/pull/32842/commits/85907c8f5f0313f173e647d39803aa2abbc088c9

then I realised it's a bit redundant ( especially for "invalid" cases ) as the "what's wrong" is covered by the `messageId` , anyways lets keep it for now. 

ty